hclasses/CTString.cpp

```diff
@@ -469,6 +469,25 @@ namespace CT {
     substringSelf(s, e + 1);
   }
 
+  void string::trimWhiteSpacesAndLinesSelf() {
+
+    int s = -1, e = privatelength;
+    const char *value = useStack ? stackValue : heapValue;
+    for (size_t j = 0; j < privatelength; j++) {
+      if (value[j] != ' ' && value[j] != '\n' && value[j] != '\r') {
+        s = j;
+        break;
+      }
+    }
+    for (size_t j = privatelength - 1; j > 0; j--) {
+      if (value[j] != ' ' && value[j] != '\n' && value[j] != '\r') {
+        e = j;
+        break;
+      }
+    }
+    substringSelf(s, e + 1);
+  }
+
```

Comment:
It hurts that we are writing stuff like this ourselves. Or copy & pasting it...

Anyway, can we merge it with the above, give that a `bool` flag that defaults to `false` that lets it also remove newline stuff?

Reply:
Moved into the same function with an optional argument.